KICS 1.4.5 & 1.4.7 is throwing exception "panic: interface conversion: interface {} is string, not map[string]interface {}"

KICS 1.4.5 & 1.4.7 is throwing exception "panic: interface conversion: interface {} is string, not map[string]interface {}"
  

### Expected Behavior

KICS should provide the scan results




### Actual Behavior
KICS throws below exception

panic: interface conversion: interface {} is string, not map[string]interface {}

goroutine 100 [running]:
github.com/Checkmarx/kics/pkg/parser/yaml.addExtraInfo({0xc000dc2bb0, 0x1, 0x1}, {0xc000a22060, 0x55})
	/home/runner/work/kics/kics/pkg/parser/yaml/parser.go:126 +0x1db
github.com/Checkmarx/kics/pkg/parser/yaml.(*Parser).Parse(0xc00018a0f0, {0xc000a22060, 0x55}, {0xc0001e1000, 0xf99, 0x1000})
	/home/runner/work/kics/kics/pkg/parser/yaml/parser.go:41 +0x4d3
github.com/Checkmarx/kics/pkg/parser.(*Parser).Parse(0xc0017660f0, {0xc000a22060, 0x55}, {0xc0001e1000, 0xf99, 0x1000})
	/home/runner/work/kics/kics/pkg/parser/parser.go:119 +0x108
github.com/Checkmarx/kics/pkg/kics.(*Service).sink(0xc00081aa10, {0x316c800, 0xc00019c008}, {0xc000a22060, 0x0}, {0x2d3ece9, 0x7}, {0x3131560, 0xc000dc2b18})
	/home/runner/work/kics/kics/pkg/kics/sink.go:34 +0x105
github.com/Checkmarx/kics/pkg/kics.(*Service).PrepareSources.func1({0x316c800, 0xc00019c008}, {0xc000a22060, 0x55}, {0x314e738, 0xc000dc2b18})
	/home/runner/work/kics/kics/pkg/kics/service.go:64 +0xa5
github.com/Checkmarx/kics/pkg/engine/provider.(*FileSystemSourceProvider).walkDir.func1({0xc000a22060, 0x55}, {0x318f608, 0xc0006c4a90}, {0x0, 0x0})
	/home/runner/work/kics/kics/pkg/engine/provider/filesystem.go:155 +0x4b5
path/filepath.walk({0xc000a22060, 0x55}, {0x318f608, 0xc0006c4a90}, 0xc00159be20)
	/opt/hostedtoolcache/go/1.17.3/x64/src/path/filepath/path.go:418 +0x125
path/filepath.walk({0xc000a18550, 0x47}, {0x318f608, 0xc0016f8b60}, 0xc00159be20)
	/opt/hostedtoolcache/go/1.17.3/x64/src/path/filepath/path.go:442 +0x28f
path/filepath.Walk({0xc000a18550, 0x47}, 0xc000a01620)
	/opt/hostedtoolcache/go/1.17.3/x64/src/path/filepath/path.go:505 +0x6c
github.com/Checkmarx/kics/pkg/engine/provider.(*FileSystemSourceProvider).walkDir(0xc000a18550, {0x316c800, 0xc00019c008}, {0xc000a18550, 0x84b6c18}, 0x0, 0x20, 0x4f00108, 0x20)
	/home/runner/work/kics/kics/pkg/engine/provider/filesystem.go:120 +0x9b
github.com/Checkmarx/kics/pkg/engine/provider.(*FileSystemSourceProvider).GetSources(0xc001721160, {0x316c800, 0xc00019c008}, 0x0, 0xc001721200, 0x0)
	/home/runner/work/kics/kics/pkg/engine/provider/filesystem.go:109 +0x1b0
github.com/Checkmarx/kics/pkg/kics.(*Service).PrepareSources(0xc00081aa10, {0x316c800, 0xc00019c008}, {0x2d3ece9, 0x7}, 0x0, 0x0)
	/home/runner/work/kics/kics/pkg/kics/service.go:60 +0x1a7
created by github.com/Checkmarx/kics/pkg/scanner.PrepareAndScan
	/home/runner/work/kics/kics/pkg/scanner/scanner.go:24 +0xe5


goroutine 38 [running]:
github.com/Checkmarx/kics/pkg/parser/yaml.addExtraInfo({0xc0012cd510, 0x1, 0x1}, {0xc00018cae0, 0x55})
	/home/runner/work/kics/kics/pkg/parser/yaml/parser.go:123 +0x1db
github.com/Checkmarx/kics/pkg/parser/yaml.(*Parser).Parse(0x310b380, {0xc00018cae0, 0x55}, {0xc000371000, 0xf99, 0x1000})
	/home/runner/work/kics/kics/pkg/parser/yaml/parser.go:38 +0x42a
github.com/Checkmarx/kics/pkg/parser.(*Parser).Parse(0xc000fffb30, {0xc00018cae0, 0x55}, {0xc000371000, 0xf99, 0x1000})
	/home/runner/work/kics/kics/pkg/parser/parser.go:110 +0xb4
github.com/Checkmarx/kics/pkg/kics.(*Service).sink(0xc0004a29a0, {0x314a780, 0xc0001a6008}, {0xc00018cae0, 0x55}, {0x2d215f5, 0x7}, {0x310f6e0, 0xc0012cd4a8})
	/home/runner/work/kics/kics/pkg/kics/sink.go:33 +0xe5
github.com/Checkmarx/kics/pkg/kics.(*Service).PrepareSources.func1({0x314a780, 0xc0001a6008}, {0xc00018cae0, 0x55}, {0x312c860, 0xc0012cd4a8})
	/home/runner/work/kics/kics/pkg/kics/service.go:64 +0xa5
github.com/Checkmarx/kics/pkg/engine/provider.(*FileSystemSourceProvider).walkDir.func1({0xc00018cae0, 0x55}, {0x316d448, 0xc000635860}, {0x0, 0x0})
	/home/runner/work/kics/kics/pkg/engine/provider/filesystem.go:151 +0x3d9
path/filepath.walk({0xc00018cae0, 0x55}, {0x316d448, 0xc000635860}, 0xc000641e20)
	/opt/hostedtoolcache/go/1.17.2/x64/src/path/filepath/path.go:418 +0x125
path/filepath.walk({0xc00073ea00, 0x47}, {0x316d448, 0xc000a9ef70}, 0xc000641e20)
	/opt/hostedtoolcache/go/1.17.2/x64/src/path/filepath/path.go:442 +0x28f
path/filepath.Walk({0xc00073ea00, 0x47}, 0xc0005f9e20)
	/opt/hostedtoolcache/go/1.17.2/x64/src/path/filepath/path.go:505 +0x6c
github.com/Checkmarx/kics/pkg/engine/provider.(*FileSystemSourceProvider).walkDir(0xc00073ea00, {0x314a780, 0xc0001a6008}, {0xc00073ea00, 0x2e0cfec8}, 0x0, 0x20, 0x49aa5b8, 0x20)
	/home/runner/work/kics/kics/pkg/engine/provider/filesystem.go:119 +0x9b
github.com/Checkmarx/kics/pkg/engine/provider.(*FileSystemSourceProvider).GetSources(0xc000d87200, {0x314a780, 0xc0001a6008}, 0x74736f7074756f22, 0xc000d872c0, 0x6f222c2273676174)
	/home/runner/work/kics/kics/pkg/engine/provider/filesystem.go:108 +0x1b0
github.com/Checkmarx/kics/pkg/kics.(*Service).PrepareSources(0xc0004a29a0, {0x314a780, 0xc0001a6008}, {0x2d215f5, 0x7}, 0x2c226c63615f6b72, 0x6574617669727022)
	/home/runner/work/kics/kics/pkg/kics/service.go:60 +0x1a7
created by github.com/Checkmarx/kics/pkg/scanner.PrepareAndScan
	/home/runner/work/kics/kics/pkg/scanner/scanner.go:24 +0xe5

(Formatted logs and samples helps us to better understand the issue)




### Steps to Reproduce the Problem

kics -s scan --no-progress --minimal-ui -q <querypath> -p <gitlab source code repository> -o <output directory> --output-name <output file>


### Specifications
(N/A if not applicable)

-   Version: 1.4.5 & 1.4.7
-   Platform: Linux 
-   Subsystem: Red Hat

--- a/pkg/parser/yaml/parser.go
+++ b/pkg/parser/yaml/parser.go
@@ -6,6 +6,7 @@ import (
 	"github.com/Checkmarx/kics/pkg/model"
 	"github.com/Checkmarx/kics/pkg/parser/utils"
 	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
 	"gopkg.in/yaml.v3"
 )
 
@@ -132,6 +133,7 @@ func addExtraInfo(documents []model.Document, filePath string) []model.Document 
 func processPlaybooks(playbooks interface{}, filePath string) {
 	sliceResources, ok := playbooks.([]interface{})
 	if !ok { // prevent panic if playbooks is not a slice
+		log.Warn().Msgf("Failed to parse playbooks: %s", filePath)
 		return
 	}
 	for _, resources := range sliceResources { // iterate over playbooks
@@ -141,14 +143,17 @@ func processPlaybooks(playbooks interface{}, filePath string) {
 
 func processPlaybooksElements(resources interface{}, filePath string) {
 	mapResources, ok := resources.(map[string]interface{})
-	if !ok { // prevent panic if playbooks is not a map
+	if !ok {
+		log.Warn().Msgf("Failed to parse playbooks elements: %s", filePath)
 		return
 	}
 	for _, value := range mapResources {
 		mapValue, ok := value.(map[string]interface{})
-		if ok {
-			processElements(mapValue, filePath)
+		if !ok {
+			log.Warn().Msgf("Failed to parse playbooks values: %s", filePath)
+			continue
 		}
+		processElements(mapValue, filePath)
 	}
 }
 

--- a/pkg/parser/yaml/parser.go
+++ b/pkg/parser/yaml/parser.go
@@ -122,18 +122,34 @@ func processElements(elements map[string]interface{}, filePath string) {
 func addExtraInfo(documents []model.Document, filePath string) []model.Document {
 	for _, documentPlaybooks := range documents { // iterate over documents
 		if playbooks, ok := documentPlaybooks["playbooks"]; ok {
-			for _, resources := range playbooks.([]interface{}) { // iterate over playbooks
-				for _, v := range resources.(map[string]interface{}) {
-					_, ok := v.(map[string]interface{})
-					if ok {
-						processElements(v.(map[string]interface{}), filePath)
-					}
-				}
-			}
+			processPlaybooks(playbooks, filePath)
 		}
 	}
 
 	return documents
+}
+
+func processPlaybooks(playbooks interface{}, filePath string) {
+	sliceResources, ok := playbooks.([]interface{})
+	if !ok { // prevent panic if playbooks is not a slice
+		return
+	}
+	for _, resources := range sliceResources { // iterate over playbooks
+		processPlaybooksElements(resources, filePath)
+	}
+}
+
+func processPlaybooksElements(resources interface{}, filePath string) {
+	mapResources, ok := resources.(map[string]interface{})
+	if !ok { // prevent panic if playbooks is not a map
+		return
+	}
+	for _, value := range mapResources {
+		mapValue, ok := value.(map[string]interface{})
+		if ok {
+			processElements(mapValue, filePath)
+		}
+	}
 }
 
 // GetCommentToken return the comment token of YAML - #

